lib/ansible/inventory/manager.py

```diff
@@ -69,9 +69,9 @@ def order_patterns(patterns):
     pattern_intersection = []
     pattern_exclude = []
     for p in patterns:
-        if p[0] == "!":
```

Comment:
instead of adding p on every condition, `if not p: continue` would shortcut the whole thing